pkg/flow/internal/controller/loader.go

```diff
@@ -77,6 +83,13 @@ func NewLoader(opts LoaderOptions) *Loader {
 		componentReg: reg,
 		workerPool:   opts.WorkerPool,
 
+		// This is a reasonable default which should work for most cases. If a component is completely stuck, we would
+		// retry and log an error every 10 seconds, at most.
+		backoffConfig: backoff.Config{
+			MinBackoff: 1 * time.Millisecond,
+			MaxBackoff: 10 * time.Second,
+		},
```

Comment:
I had a look at the backoff config, and I think we need to set `MaxRetries` to something other than 0 for `Ongoing()` to ever return `false`. Otherwise I think it'll keep retrying infinitely. IMO there should be a retry limit, but I don't know what's a reasonable one.

Reply:
This is actually intentional - we will be retrying forever to ensure we don't drop any data.
The `Ongoing()` will stop when the parent `context` is cancelled, which is when the `Flow` controller shuts down when the agent is shutting down. 

So we have infinite retry loop that can end only if: a) the worker pool is unclogged somehow, b) the agent shuts down.